doc/ebpf/env.rst

```diff
@@ -0,0 +1,630 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Send a module for testing to the target:
+
+.. code-block:: bash
+
+    # For user level network interface
+    rsync linux/build/lib/test_bpf.ko arc:/root
+
+    # For TUN/TAP network interface
+    rsync linux/build/lib/test_bpf.ko arc-tap:/root
+
+Run the module on the target:
+
+.. code-block:: bash
+
+    # Run all tests
+    insmod test_bpf.ko
+
+    # Run a specific
+    insmod test_bpf.ko test_id=42
+
+    # Run a range of tests
+    insmod test_bpf.ko test_range=42,142
+
+Building and Running eBPF Programs
+----------------------------------
+
+.. warning::
+
+    Old operations systems like CentOS 7 and Ubuntu 18.04 contain
+    old versions of ``clang`` which may be not sufficient for building
+    modern eBPF programs. If building eBPF programs fails then try to
+    build the latest ``clang`` with eBPF target following
+    :ref:`a corresponding guide <ebpf-clang>` and put it into ``PATH``.
+
+Testbench contains a bunch of examples of eBPF programs. You
+can build and load them using these commands from the root directory
+of the testbench:
+
+.. code-block:: bash
+
+    # Build dependencies
+    make
+
+    # Load programs for user level network interface
+    make qemu-load
+
+    # or for TUN/TAP network interface
+    make USE_TAP=1 qemu-load
+
+Run a program:
+
+.. code-block:: bash
+
+    # Manually on target's side
+    ./minimal
+
+    # ... or using testbench on host's side
+    make run-minimal
+
+Explore ``README.md`` for `ARC eBPF Testbench <https://github.com/foss-for-synopsys-dwc-arc-processors/arc-bpf-testbench>`_
+or run ``make help`` for information about available commands.
+
+Using NFS for Building eBPF Programs Right on the Target
+--------------------------------------------------------
+
+Configuring NFS in CentOS 7 or Fedora
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Install NFS to the host:
+
+.. code-block:: bash
+
+    # For CentOS 7
+    sudo yum install nfs-utils
+
+    # For Fedora
+    sudo dnf install nfs-utils
+
+Enable services and add rules for firewall:
+
+.. code-block:: bash
+
+    sudo systemctl enable --now rpcbind nfs-server
+    sudo firewall-cmd --add-service=nfs --permanent
+    sudo firewall-cmd --reload
+
+    # Optional (only if you are going to use SSHFS instead of NFS)
+    sudo systemctl enable sshd
+    sudo systemctl start sshd
+
+Configuring NFS in Ubuntu 18.04
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Install NFS to the host and enable it:
+
+.. code-block:: bash
+
+    sudo apt install nfs-kernel-server
+    sudo systemctl enable --now nfs-server
+
+Configuring ``/etc/exports``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Add this line to ``/etc/exports`` (you can find ``anonuid`` and ``anongid``
+for your user using ``id -u`` and ``id -g`` respectively)::
+
+    /nfs *(rw,all_squash,anonuid=1000,anongid=1000,no_subtree_check,insecure)
+
+Update the table of exported NFS file systems:
+
+.. code-block:: bash
+
+    sudo exportfs -rrv
```

Comment:
Is the double `r` intentional? The man page on my systems says nothing about it. I also append `-a`. Not sure if it would help though.